client/src/navigation/MainNavigator.js

```diff
@@ -43,11 +44,21 @@ const MainTabNavigator = TabNavigator(
   {
     Home: {
       screen: HomeNavigator,
-      navigationOptions: {
+      navigationOptions: ({ navigation }) => ({
         tabBarLabel: 'Home',
         // eslint-disable-next-line react/prop-types
         tabBarIcon: ({ tintColor }) => <Icon size={34} name="home" color={tintColor} />,
-      },
+        tabBarOnPress() {
+          const resetAction = NavigationActions.reset({
+            index: 0,
+            key: 'Home',
+            actions: [
+              NavigationActions.navigate({ routeName: 'Root' }),
+            ],
+          });
+          navigation.dispatch(resetAction);
```

Comment:
did you try ditching the reset and just using `navigation.navigate('Root');`?

Reply:
it might help if you explain what you tried before you came to this being the best option

Reply:
This was the first thing I tried, and its what 'the internet' documents as the correct way to reset a stack.

There is currently nothing in our code to reset the stack and nothing documented in stacknav that would suggest it should be resetting. When we swap tabs it should just keep the set of tabs in that stack in place. 

#448

Reply:
If you just try nav to root you will end up with a weirdly large stack of screens in your stack i believe as it would just add another home screen to the mix.